bokehjs/src/coffee/core/selection_manager.ts

```diff
@@ -10,47 +10,152 @@ import {DataSource} from "models/sources/data_source"
 export type Renderer = any
 export type RendererView = any
 
+export type HitTestResult = Selection | null
+
+export abstract class SelectionPolicy extends Model {
+
+  hit_test(geometry: Geometry, renderer_views: RendererView[]): HitTestResult{
+    return null
+  }
+
+  do_selection(hit_test_result: HitTestResult | null, renderer_views: RendererView[], final: boolean, append: boolean): boolean {
```

Comment:
`HitTestResult` is defined as `Selection | null`, so this additional `null` is unnecessary.